src/python/dxpy/executable_builder.py

```diff
@@ -152,18 +179,18 @@ def verify_developer_rights(prefixed_name):
     return FoundExecutable(name=name_without_prefix, version=version, id=executable_id)
 
 
-def assert_consistent_regions(from_spec, from_command_line, builder_exception):
+def assert_consistent_regions(from_spec, from_command_line, executable_builder_exception):
     """
     Verifies the regions passed with --region CLI argument and the ones
     specified in regionalOptions are the same (if both CLI and spec were used)
     """
     if from_spec is None or from_command_line is None:
         return
     if set(from_spec) != set(from_command_line):
```

Comment:
I reverted my previous chanage, which only check if --region is a subset of regionalOptions from the json spec. It sounds more consistent with the helpstring `help="Enable the app/globalworkflow in this region. This flag can be specified multiple times to enable the app/globalworkflow in multiple regions. If --region is not specified, then the enabled region(s) will be determined by 'regionalOptions' in dxapp.json, or the project context."` , but I don't want to change the behavior here since I am not sure about the side effect when building app/gwf.

Reply:
I don't see a problem with leaving the original behavior here. The way I read it, you are meant to either use `--region` without restrictions if the JSON doesn't contain `regionalOptions`, or not use the flag if `regionalOptions` are present. In case both are used at the same time, it enforces that they match exactly to avoid ambiguity.

Reply:
After some consideration and discussions with Yuxin (and her investigation), we've decided it's best to tackle `--region` separately and its behavior won't be changed here.